test/e2e/service_test.go

```diff
@@ -75,54 +74,58 @@ func (data *TestData) testClusterIP(t *testing.T, isIPv6 bool, clientNamespace,
-	defer cleanupFunc()
 	t.Run("HostNetwork Endpoints", func(t *testing.T) {
+		_, _, cleanupFunc := createAndWaitForPod(t, data, data.createNginxPodOnNode, hostNginx, nodeName(0), serverNamespace, true)
+		defer cleanupFunc()
 		skipIfNamespaceIsNotEqual(t, serverNamespace, data.testNamespace)
-		testClusterIPCases(t, data, url, clients, hostNetworkClients, clientNamespace)
+		time.Sleep(time.Second)
+		testClusterIPCases(t, data, url, clients, hostNetworkClients, clientNamespace, Connected)
 	})
 }
 
-func testClusterIPCases(t *testing.T, data *TestData, url string, clients, hostNetworkClients map[string]string, namespace string) {
-	t.Run("All Nodes can access Service ClusterIP", func(t *testing.T) {
+func testClusterIPCases(t *testing.T, data *TestData, url string, clients, hostNetworkClients map[string]string, namespace string, expectedConnectivity PodConnectivityMark) {
+	t.Run("Connect to Service ClusterIP from Node", func(t *testing.T) {
 		skipIfProxyAllDisabled(t, data)
 		skipIfKubeProxyEnabled(t, data)
 		skipIfNamespaceIsNotEqual(t, namespace, data.testNamespace)
 		for node, pod := range hostNetworkClients {
-			testClusterIPFromPod(t, data, url, node, pod, true, namespace)
+			testClusterIPFromPod(t, data, url, node, pod, true, namespace, expectedConnectivity)
 		}
 	})
-	t.Run("Pods from all Nodes can access Service ClusterIP", func(t *testing.T) {
+	t.Run("Connect to Service ClusterIP from Pod", func(t *testing.T) {
 		for node, pod := range clients {
-			testClusterIPFromPod(t, data, url, node, pod, false, namespace)
+			testClusterIPFromPod(t, data, url, node, pod, false, namespace, expectedConnectivity)
 		}
 	})
 }
 
-func testClusterIPFromPod(t *testing.T, data *TestData, url, nodeName, podName string, hostNetwork bool, namespace string) {
-	cmd := []string{"/agnhost", "connect", url, "--timeout=1s", "--protocol=tcp"}
-	err := wait.PollImmediate(1*time.Second, 5*time.Second, func() (bool, error) {
-		t.Logf(strings.Join(cmd, " "))
-		stdout, stderr, err := data.RunCommandFromPod(namespace, podName, agnhostContainerName, cmd)
-		t.Logf("stdout: %s - stderr: %s - err: %v", stdout, stderr, err)
-		if err == nil {
-			return true, nil
+func testClusterIPFromPod(t *testing.T, data *TestData, url, nodeName, podName string, hostNetwork bool, namespace string, expectedConnectivity PodConnectivityMark) {
+	cmd := fmt.Sprintf("for i in $(seq 1 3); do /agnhost connect %s --timeout=1s --protocol=tcp; done;", url)
+	stdout, stderr, err := data.RunCommandFromPod(namespace, podName, agnhostContainerName, []string{"sh", "-c", cmd})
+	connectivity := Connected
+	if err != nil || stderr != "" {
+		// log this error as trace since may be an expected failure
+		log.Tracef("Pod '%s' on Node '%s' (hostNetwork: %t) to %s: error when running command: err - %v /// stdout - %s /// stderr - %s", podName, nodeName, hostNetwork, url, err, stdout, stderr)
```

Comment:
It already knows the expected result, and should just log with `t.Log` when `connectivity` doesn't match `expectedConnectivity`

Reply:
Thanks, updated.